lib/_http_incoming.js

```diff
@@ -67,6 +69,8 @@ IncomingMessage.prototype.setTimeout = function(msecs, callback) {
 
 
 IncomingMessage.prototype.read = function(n) {
+  if (!this._consuming)
+    this._readableState.readingMore = false;
```

Comment:
Out of curiosity, would there be a behavioral difference if you dropped the if statement and wrote it as `this._readableState.readingMore = this._consuming;`?

Also, this could use a comment explaining why it's necessary to fiddle with `this._readableState` and why `this._consuming` should be true when monkey-patching `this.read.`.